students/jesse_miller/session09/cli_main.py

```diff
@@ -0,0 +1,180 @@
+'''
+
+class MailRoom:
+    '''
+    All the calls and commands will be through this class.  Manipulation of
+    donor information is done in donor_models.py, and all of the execution for
+    this subprogram is from mailroom_oo.py
+    '''
+    @staticmethod
+    def list_donors():
+        """
+        Lists existing donors.
+        """
+        print(f"\n{'-'*15}\nList of Donors:\n{'-'*15}")
+        current = alms.list_donor()
+        for donor in current:
+            print(donor)
+        print(f"{'-'*15}\n")
+
+
+    @staticmethod
+    def print_report():
+        """
+        Prints a summary report of all donors.
+        """
+        headers = ["Donor Name", "Total Given", "Times Donated", "Average Gift"]
+        print(f"\n{'-'*80}\n{{:17}} | {{:<19}} | {{:<15}} | {{:<19}}\n{'-'*80}"\
+        .format(headers[0], headers[1], headers[2], headers[3]))
+
+        donor_data = alms.create_report()
+        for row in donor_data:
+            print('{:17} | ${:<18,.2f} | {:<15} | ${:<16,.2f}'.format\
+            (row[0], row[1], row[2], row[3]))
+        print(f"{'-'*80}\n")
+
+
+    @staticmethod
+    def donor_mail_choice():
+        '''
+        This section allows the user to mail a donor
+        '''
+        donor = ''
+        MailRoom.list_donors()
+        try:
+            donor = str(input('Who would you like to mail (all for all): '))
+            if alms.find_donor(donor):
+                MailBox.mail_send(donor)
+            if donor == 'all':
+                MailBox.mail_send_all()
+        except (KeyboardInterrupt, EOFError, ValueError):
+            MailRoom.safe_input()
+
+
+    @staticmethod
+    def safe_input():
+        '''
+        This will be for handling keyboard exceptions
+        '''
+        return None
+
+
+    @staticmethod
+    def goodbye():
+        '''
+        Gracefully exits
+        '''
+        print('Goodbye!')
+        sys.exit()
+
+
+    @staticmethod
+    def donor_del():
+        '''
+        This section allows the user to delete a donor
+        '''
+        try:
+            MailRoom.list_donors()
+            del_donor = str(input('Enter the name of the donor to remove: '))
+            del alms.donors_dict[del_donor]
+            MailRoom.list_donors()
+        except (KeyboardInterrupt, EOFError, ValueError):
+            MailRoom.safe_input()
+
+
+    @staticmethod
+    def donor_add():
+        '''
+        This section allows the user to add a donor
+        '''
+        try:
+            MailRoom.list_donors()
+            donor = str(input('Enter the name of the donor to add: '))
+            alms.donor_creation(donor)
+            while True:
+                d_num = int(input('How many donations were made: '))
+                while d_num > 0:
+                    donations = float(input('Enter their donation: '))
+                    #self.donors_dict[donor].append(new_don)
+                    tools.donation_add(donations)
```

Comment:
You need to call _donations_add_ from the new Donor that you have created, this line should be:

```
alms.donors_dict[donor].donation_add(donations)
```

Reply:
I had tried that once, this is what I get:

'Kerry': <donor_models.DonorTools object at 0x1063d6d30>}